src/BloomExe/Browser.cs

```diff
@@ -419,21 +419,23 @@ private void PasteFilteredText(bool removeSingleLineBreaks)
 
 		void OnShowContextMenu(object sender, GeckoContextMenuEventArgs e)
 		{
+			MenuItem FFMenuItem = null;
 			Debug.Assert(!InvokeRequired);
 			if (ContextMenuProvider != null)
 			{
-				ContextMenuProvider(e);
-#if DEBUG
-				e.ContextMenu.MenuItems.Add("Open Page in Firefox (which must be in the PATH environment variable)", new EventHandler(OnOpenPageInSystemBrowser));
-#endif
-				if (ContextMenuProvider(e))
-					return; // only the provider's items
+				var replacesStdMenu = ContextMenuProvider(e);
+				FFMenuItem = e.ContextMenu.MenuItems.Add("Open Page in Firefox (which must be in the PATH environment variable)",
+					new EventHandler(OnOpenPageInSystemBrowser));
 
+				if (replacesStdMenu)
+					return; // only the provider's items
 			}
 			var m = e.ContextMenu.MenuItems.Add("Edit Stylesheets in Stylizer", new EventHandler(OnOpenPageInStylizer));
 			m.Enabled = !string.IsNullOrEmpty(GetPathToStylizer());
 
-			e.ContextMenu.MenuItems.Add("Open Page in Firefox (which must be in the PATH environment variable)", new EventHandler(OnOpenPageInSystemBrowser));
+			if(FFMenuItem == null)
+				e.ContextMenu.MenuItems.Add("Open Page in Firefox (which must be in the PATH environment variable)",
+					new EventHandler(OnOpenPageInSystemBrowser));
```

Comment:
I'd like to make a method out of this couple of lines of duplicated code and strings. Is there a more generic name than "AddOpenPageInFFItem"? The essence seems to be adding menu items that are mainly for developers...how about "AddHtmlDebuggingMenuItems"?
Not sure it's worth another rev (and the original had the same problem)...but in case you're doing another anyway...